src/Microsoft.XmlSerializer.Generator/src/Sgen.cs

```diff
@@ -481,5 +489,80 @@ private static string GetTempAssemblyName(AssemblyName parent, string ns)
         {
             return parent.Name + ".XmlSerializers" + (ns == null || ns.Length == 0 ? "" : "." + ns.GetHashCode());
         }
+
+        private static void ParseReferences(string value, Dictionary<string, string> dictionary)
+        {
+            var referencelist = new List<string>();
+            if (value.Length > 0)
+            {
+                string[] entries = value.Split(new char[] { ';' });
+                for (int i = 0; i < entries.Length; i++)
+                {
+                    string entry = entries[i].Trim();
+                    if (string.IsNullOrEmpty(entry))
+                        continue;
+                    referencelist.Add(entry);
+                }
+            }
+
+            foreach (var reference in referencelist)
+            {
+                if (reference.EndsWith(".dll") || reference.EndsWith(".exe"))
+                {
+                    if (File.Exists(reference))
+                    {
+                        string filename = Path.GetFileNameWithoutExtension(reference);
+                        if (!string.IsNullOrEmpty(filename))
+                        {
+                            dictionary.Add(filename, reference);
+                        }
+                    }
+                }
+
+            }
+        }
+
+        private static Assembly SgenAssemblyResolver(object source, ResolveEventArgs e)
+        {
+            try
+            {
+                if (string.IsNullOrEmpty(s_references) || string.IsNullOrEmpty(e.Name) || e.Name.Split(',').Count() == 0)
```

Comment:
>.Count() [](start = 107, length = 8)

String.Split returns string[] which has a .Length property. Count() is a Linq extension method on IEnumerable (which array implements by convention) which is overkill.